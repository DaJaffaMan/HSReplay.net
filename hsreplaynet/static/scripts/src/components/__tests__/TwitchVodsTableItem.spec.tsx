import TwitchVodsTableItem from "../TwitchVodsTableItem";
import React from "react";
import renderer from "react-test-renderer";

describe("TwitchVodsTableItem", () => {
	test("renders correctly", () => {
		Date.now = jest.fn(() => new Date("2018-10-22T20:14:00Z").getTime());
		const component = renderer.create(
			<TwitchVodsTableItem
				rank={0}
				legendRank={1234}
				channelName={"HearthSim"}
				won
				wentFirst
				gameLengthSeconds={1337}
				gameDate={new Date("2018-10-22T20:13:30Z")}
				opposingPlayerClass={"MAGE"}
				opposingArchetype={null}
				gameType={"BGT_RANKED_STANDARD"}
				cardData={null as any}
			/>,
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
