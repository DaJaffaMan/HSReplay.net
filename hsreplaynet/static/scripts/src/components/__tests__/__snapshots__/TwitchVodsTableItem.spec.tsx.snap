// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitchVodsTableItem renders correctly 1`] = `
Array [
  <span
    className="twitch-vod-table-matchup"
  >
    <div
      className="tooltip-wrapper"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onTouchStart={[Function]}
    >
      <aside
        className="sr-only"
      >
        <p>
          Legend 1234
        </p>
      </aside>
      <figure
        className="rank-icon-standalone"
      >
        <img
          alt="Legend 1234"
          src="/static/images/ranked-medals/Medal_Ranked_Legend.png"
        />
        <figcaption>
          1234
        </figcaption>
      </figure>
    </div>
    <span
      className="text-twitch twitch-vod-channel-name"
    >
      <img
        alt="Twitch"
        src="/static/images/socialauth/twitch.png"
      />
       
      <span>
        HearthSim
      </span>
    </span>
    <img
      className="vs-icon"
      src="/static/images/vs.png"
    />
    <div
      className="tooltip-wrapper"
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onTouchStart={[Function]}
    >
      <aside
        className="sr-only"
      >
        Mage
      </aside>
      <img
        className="class-icon"
        src="/static/images/64x/class-icons/mage.png"
      />
    </div>
    <span
      className="opposing-archetype"
    >
      Mage
    </span>
  </span>,
  <p>
    <time
      dateTime="2018-10-22T20:13:30.000Z"
    >
      1 minute ago
    </time>
    <strong
      className="twitch-vod-table-outcome"
    >
      Won
    </strong>
     
    going 
    <strong>
      <img
        className="twitch-vod-table-inline-icon"
        src="/static/images/first.png"
      />
       
      first
    </strong>
     
    after
     
    <span
      className="game-duration"
    >
      <span
        className="glyphicon glyphicon-time"
      />
       
      22.3 min
    </span>
  </p>,
]
`;
